packages/react-static-tweets/src/twitter.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { createContext, ReactNode, useContext } from 'react'
+import { ConfigInterface } from 'swr'
+
+// TODO: make this more specific
+export type TweetAst = Array<any>
+
+export type TwitterContextValue = {
+  // static tweet ast info
+  tweetAstMap: TweetAstMap
+
+  // SWR config for dynamically fetching tweet ast info
+  swrOptions: ConfigInterface
+}
+
+export type TweetAstMap = {
+  [tweetId: string]: TweetAst
+}
+
+// Saves the tweets returned as props to the page
+const TwitterContext = createContext<TwitterContextValue>({
```

Comment:
i always default to exporting more than necessary just in case people want access to the original values

Reply:
Yeah, I get that, but wouldn't it be confusing for users to have two providers that behave differently tho (`TwitterContextProvider` and `TwitterContext.Provider`)?

What if we re-export the intended (overriding) provider + consumer hook from the module namespace directly (e.g. `import { TwitterContextProvider } from 'react-static-tweets'`) so that the distinction is a little clearer?

I think material-ui does something like this. [Related docs](https://material-ui.com/customization/theming/)

Reply:
I think this could be important, because the context is where the tweets AST lives, so the actual (non-overridable) context should really only be used by the package code itself, or by people who know what they're doing and who don't mind using unstable/undocumented internals (for some advanced use case or whatever).

Confused users could override the context and wipe out the tweet data without an obvious reason as to why this is happening.

What do you think?

Reply:
i'm fine with either way.

i tend to default to exporting more, but i can see the argument either way. let's leave as-is for now & ship it :)